benchmarks/benchmarks/bench_linalg.py

```diff
@@ -217,3 +217,40 @@ def time_transpose(self, shape, npdtypes):
 
     def time_vdot(self, shape, npdtypes):
         np.vdot(self.xarg, self.x2arg)
+
+
+class MatmulStrided(Benchmark):
+    # some interesting points selected from
+    # https://github.com/numpy/numpy/pull/23752#issuecomment-2629521597
+    # (m, p, n, batch_size)
+    args = [
+        (2, 2, 2, 1), (2, 2, 2, 10), (5, 5, 5, 1), (5, 5, 5, 10),
+        (10, 10, 10, 1), (10, 10, 10, 10), (20, 20, 20, 1), (20, 20, 20, 10),
+        (50, 50, 50, 1), (50, 50, 50, 10),
+        (150, 150, 100, 1), (150, 150, 100, 10),
+        (400, 400, 100, 1), (400, 400, 100, 10)
+    ]
+
+    param_names = ['configuration']
+
+    def __init__(self):
+        args = self.args
+
+        a1raw_a2 = [[np.random.rand(batch_size * m * 2 * n).reshape(
+            (batch_size, m, 2 * n)
+        ), np.random.rand(batch_size * n * p).reshape(
+            (batch_size, n, p)
+        )] for [m, p, n, batch_size] in args]
+
+        self.args_map = dict([[
+            'matmul_' + 'm%03d_p%03d_n%03d_bs%02d' % args[i],
+            [a1raw_a2[i][0][:, :, ::2], a1raw_a2[i][1]]
+        ] for i in range(len(self.args))])
```

Comment:
```suggestion
        self.args_map = {
            'matmul_' + 'm%03d_p%03d_n%03d_bs%02d' % args[i]:
                [a1raw_a2[i][0][:, :, ::2], a1raw_a2[i][1]]
            for i in range(len(self.args))})
```
or somethingi like this, I likey misformatted something.

Reply:
Another idea came to my mind and I moved a lot from `__init__` into the `setup` function. The code is much cleaner now.